add concept entry for postgresql transactions

christina-ml opened the pull request:
<!--
👋 Hi, thanks for submitting a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

Add New Concept Entry for PostgreSQL Transactions.

### Issue Solved

<!--
Please use the following format(s) to link the issue numbers that your contribution addresses:
Closes #issue-number (where `issue-number` corresponds to the PRs' respective issue)
-->
Closes #4647 

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
```diff
@@ -0,0 +1,101 @@
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+  - 'An nth subject name'
```
The subject name should be - 'Developer Tools'

Radhika-okhade replied:
```suggestion
  - 'Developer Tools'
```

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
```diff
@@ -0,0 +1,101 @@
+  - 'Database'
+  - 'Finance'
+CatalogContent:
+  - 'paths/design-databases-with-postgresql'
+  - 'getting-started-off-platform-for-data-science'
+  - 'ext-courses/google-associate-cloud-engineer-configuring-google-cloud'
```
```suggestion
  - 'paths/data-science'
  - 'paths/data-science-foundations'
```

christina-ml replied:
Replacing paths with suggested Data Science paths.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+
+`BEGIN` starts a transaction block.
+
+```sql
````
```suggestion
```pseudo
```

christina-ml replied:
Remove extra words.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+
+`ROLLBACK` is used to undo a transaction. It cancels all changes made in the current transaction block. This can also be used if partway through the transaction, there is no need to commit these changes.
+
+```sql
````
```suggestion
```pseudo
```

christina-ml replied:
Correct syntax highlighting for SQL does not occur using `pseudo`.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+
+Putting it all together, here is the general syntax for the three commands:
+
+```sql
````
```suggestion
```pseudo
```

christina-ml replied:
Change `sql` to `pseudo`.

christina-ml replied:
Correct syntax highlighting for SQL does not occur using `pseudo`.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+
+or
+
+```sql
````
```suggestion
```pseudo
```

christina-ml replied:
Change `sql` to `pseudo`.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+-- set of statements
+COMMIT;
+```
+
````
Please add an output block for the above example

christina-ml replied:
Add output block with description for the example.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
```diff
@@ -0,0 +1,101 @@
+
+To have more control over statements in a transaction, the use of the `SAVEPOINT` command allows a savepoint to be defined. After defining a savepoint, if needed, the transaction can be rolled back to the savepoint with the `ROLLBACK TO` command.
+
+Here is an oversimplified example that demonstrates the usage of a transaction block, implementing the `SAVEPOINT` and `ROLLBACK TO` commands, giving more control over the statements within the transaction block:
```
Oversimplified is overused in the doc. It can be removed here. 
```suggestion
Here is an example that demonstrates the usage of a transaction block, implementing the `SAVEPOINT` and `ROLLBACK TO` commands, giving more control over the statements within the transaction block:
```

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+  WHERE name = 'Carol';
+COMMIT;
+```
+
````
Please add output block for the above example

christina-ml replied:
Add outputs with explanations for the example.

Radhika-okhade commented:
Hey @christina-ml  thank you for contributing to Codecademy Docs! The entry is very well written! 😄

I've suggested some minor changes, could you please review and modify those at your earliest convenience? Thank you! 😃

christina-ml commented on content/postgresql/concepts/transactions/transactions.md:
```diff
@@ -0,0 +1,101 @@
+
+## Syntax
+
+There are three main commands in a transaction. These are the `BEGIN`, `COMMIT`, and `ROLLBACK` statements.
```
Remove extra words.

christina-ml commented on content/postgresql/concepts/transactions/transactions.md:
````diff
@@ -0,0 +1,101 @@
+
+`COMMIT` permanently saves the changes made within the transaction block to the database.
+
+```sql
````
Change `sql` to `pseudo`

christina-ml replied:
Correct syntax highlighting for SQL does not occur using `pseudo`.

christina-ml commented:
Hey @Radhika-okhade thank you! I modified the changes based on your suggestions, and adding the outputs to the examples.

I tried changing the code blocks to use `pseudo`, but it did not highlight the SQL syntax correctly, so I reverted it back to `sql`.

Radhika-okhade commented:
Hey! @christina-ml Can you please output for both Example 1 and Example 2.

christina-ml commented:
Hey @Radhika-okhade! Could you please clarify what you mean by that? I have updated the file to include the outputs for Example 1 and Example 2.

Radhika-okhade commented on content/postgresql/concepts/transactions/transactions.md:
```diff
@@ -0,0 +1,128 @@
+---
+Title: 'Transactions'
+Description: Transactions bundle one or more steps into a single unit of work; an all-or-nothing operation where all statements succeed or all statements fail.'
```
```suggestion
Description: 'Transactions bundle one or more steps into a single unit of work; an all-or-nothing operation where all statements succeed or all statements fail.'
```

Sriparno08 reviewed (approved):
Looks good, @christina-ml!
